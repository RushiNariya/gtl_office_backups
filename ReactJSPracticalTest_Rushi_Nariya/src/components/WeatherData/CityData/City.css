.city {
  display: flow-root;
  font-family: 'Open Sans', sans-serif;
  text-align: left;
  width: 100%;
  background: rgba(255, 255, 255, 0.3);
  border-radius: 4px;
  margin: 10px auto;
  padding: 10px 35px 10px 15px;
  box-sizing: border-box;
  box-shadow: 0 1/8rem 1/8rem 0 rgba(0, 0, 0, 0.05);
  cursor: pointer;
  -webkit-tap-highlight-color: rgba(8, 7, 7, 0);
  -webkit-tap-highlight-color: transparent;
  transition: all 0.3s ease-in-out;
  position: relative;
}

@media (min-width: 768px) {
  .city {
    padding: 10px 45px 10px 25px;
  }
}

.city::after {
  content: '';
  clear: both;
  display: table;
}

city:hover {
  border-color: #ddd;
  box-shadow: 0 1/8rem 1/8rem 0 rgba(0, 0, 0, 0.12);
}
@media (min-width: 768px) {
  city:hover {
    transform: scale(1.05);
  }
}

.city .column-1 {
  display: inline-block;
  float: left;
  width: calc(100% - 60px);
  text-align: left;
}

.city .column-2 {
  display: inline-block;
  float: right;
  width: 60px;
  text-align: right;
}

.city .name {
  font-size: 18px;
  font-weight: 700;
  color: #fff;
  opacity: 1;
}

@media (min-width: 760px) {
  .city .name {
    font-size: 36px;
  }
}

@media (min-width: 460px) {
  .city .name {
    font-size: 26px;
  }
}

@media (max-width: 460px) {
  .city .name {
    font-size: 20px;
  }
}

.flag {
  display: inline-block;
  vertical-align: middle;
  margin-bottom: 2px;
}

@media (max-width: 460px) {
  .flat {
    height: 22px;
    width: 22px;
  }
}

.descr {
  font-size: 16px;
}

@media (max-width: 460px) {
  .descr {
    font-size: 14px;
  }
}

.temp {
  position: relative;
}
.temp .units {
  font-size: 20px;
  color: #fff;
  line-height: 80px;
  font-weight: 600;
}

@media (max-width: 460px) {
  .temp .units {
    font-size: 16px;
    line-height: 80px;
  }
}
.temp .metrics {
  font-family: 'Roboto', 'Avenir', Helvetica, Arial, sans-serif;
  width: 21px;
  color: #fff;
  font-size: 17px;
  font-weight: 300;
  position: absolute;
  top: 25px;
}

.cancel-button {
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  outline: none;
  text-decoration: none;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  -webkit-tap-highlight-color: transparent;
  background: transparent;
  cursor: pointer;
  -webkit-appearance: none;
  position: absolute;
  right: 7px;
  top: 0px;
}

.cancel-button .icon-cancel {
  font-size: 24px;
  color: #000;
  opacity: 0.5;
  transition: all 0.3s ease-in-out;
}

.cancel-button:hover .icon-cancel {
  opacity: 1;
}

.modal {
  width: 100%;
  height: 100%;
}
.modal-dialog {
  max-width: 100% !important;
}
.modal-content {
  background-image: url('../../../assets/jungle.jpg');
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
}
