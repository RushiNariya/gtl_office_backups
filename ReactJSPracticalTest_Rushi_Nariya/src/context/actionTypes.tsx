export const SEARCH_BY_CITY = 'SEARCH_BY_CITY';
export const SET_WEATHER_DATA = 'SET_WEATHER_DATA';
export const SEARCH_CITY = 'SEARCH_CITY';
export const REMOVE_CITY = 'REMOVE_CITY';
export const SET_ERROR = 'SET_ERROR';
export const RESET_ERROR = 'RESET_ERROR';
export const REFRESH_STATE = 'REFRESH_STATE';