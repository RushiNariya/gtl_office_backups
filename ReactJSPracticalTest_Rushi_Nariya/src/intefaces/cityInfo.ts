interface cityInfo{
  name: string;
  country: string;
  temp: number;
  icon: string;
  latitude: number;
  longitude: number;
  minTemp: number;
  maxTemp: number;
}

export default cityInfo;