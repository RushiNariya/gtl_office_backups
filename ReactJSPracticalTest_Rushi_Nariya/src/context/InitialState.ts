/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-empty-function */

import { createContext } from "react";
import cityInfo from "../intefaces/cityInfo";
import weatherInfo from "../intefaces/weatherInfo";

export type initState = {
  data: weatherInfo;
  city: Array<cityInfo>;
  error: string;
  searchByCityHandler: (data: weatherInfo) => void;
  searchCityHandler: (data: cityInfo) => void;
  setWeatherDataHandler: (data: weatherInfo) => void;
  errorHandler: (error: string) => void;
  resetSetErrorHandler: () => void;
  removeCityHandler: (city: string) => void;
};
  
const initialState: initState = {
  data: {},
  city: [],
  error: '',
  searchByCityHandler: (data) => {},
  searchCityHandler: (data) => {},
  setWeatherDataHandler: (data) => {},
  errorHandler: (error) => {},
  resetSetErrorHandler: () => {},
  removeCityHandler: (city) => {},
};

export const GlobalContext = createContext(initialState);

export default initialState;