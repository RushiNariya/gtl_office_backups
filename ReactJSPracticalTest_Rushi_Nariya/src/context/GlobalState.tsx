/* eslint-disable @typescript-eslint/no-non-null-assertion */
/* eslint-disable @typescript-eslint/no-empty-function */
import React, { useReducer, useEffect } from 'react';
import PropTypes from 'prop-types';
import AppReducer from './reducer';
import {
  searchByCityAction,
  searchCityAction,
  setWeatherDataAction,
  setErrorAction,
  resetErrorAction,
  removeCityAction,
  refreshStateAction,
} from './actions';
import weatherInfo from '../intefaces/weatherInfo';
import cityInfo from '../intefaces/cityInfo';
import { GlobalContext, initState } from './InitialState';


type props = {
  children: any;
  initialState: initState;
};

export const GlobalProvider = ({ children, initialState }: props): JSX.Element => {
  const [state, dispatch] = useReducer(AppReducer, initialState);

  function searchByCityHandler(data: weatherInfo): void {
    dispatch(searchByCityAction(data));
  }
  function searchCityHandler(city: cityInfo): void {
    dispatch(searchCityAction(city));
  }
  function removeCityHandler(city: string): void {
    dispatch(removeCityAction(city));
  }
  function setWeatherDataHandler(data: weatherInfo): void {
    dispatch(setWeatherDataAction(data));
  }
  function errorHandler(error: string): void {
    dispatch(setErrorAction(error));
  }
  function resetSetErrorHandler(): void {
    dispatch(resetErrorAction());
  }

  function refreshState(): void {
    const refreshedState: Array<cityInfo> = localStorage.getItem('state')
      ? JSON.parse(localStorage.getItem('state')!)
      : initialState.city;
    dispatch(refreshStateAction(refreshedState));
  }

  useEffect(() => {
    refreshState();
  }, []);

  useEffect(() => {
    localStorage.setItem('state', JSON.stringify(state.city));
  }, [state]);

  return (
    <GlobalContext.Provider
      value={{
        data: state.data,
        city: state.city,
        error: state.error,
        searchByCityHandler,
        searchCityHandler,
        setWeatherDataHandler,
        errorHandler,
        resetSetErrorHandler,
        removeCityHandler,
      }}
    >
      {children}
    </GlobalContext.Provider>
  );
};

GlobalProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};
