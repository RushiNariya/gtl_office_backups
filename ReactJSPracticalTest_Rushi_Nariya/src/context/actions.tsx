import cityInfo from '../intefaces/cityInfo';
import weatherInfo from '../intefaces/weatherInfo';
import {
  REFRESH_STATE,
  REMOVE_CITY,
  RESET_ERROR,
  SEARCH_BY_CITY,
  SEARCH_CITY,
  SET_ERROR,
  SET_WEATHER_DATA,
} from './actionTypes';
import { initState } from './InitialState';

export interface ActionMethodType {
  type: string;
  payload: weatherInfo | string | initState | Array<cityInfo>;
}

export const searchByCityAction = (data: weatherInfo): ActionMethodType => ({
  type: SEARCH_BY_CITY,
  payload: data,
});

export const setWeatherDataAction = (data: weatherInfo): ActionMethodType => ({
  type: SET_WEATHER_DATA,
  payload: data,
});

export const searchCityAction = (data: cityInfo): ActionMethodType => ({
  type: SEARCH_CITY,
  payload: data,
});

export const removeCityAction = (city: string): ActionMethodType => ({
  type: REMOVE_CITY,
  payload: city,
});

export const setErrorAction = (error: string): ActionMethodType => ({
  type: SET_ERROR,
  payload: error,
});

export const resetErrorAction = (): ActionMethodType => ({
  type: RESET_ERROR,
  payload: '',
});

export const refreshStateAction = (
  state: Array<cityInfo>
): ActionMethodType => ({
  type: REFRESH_STATE,
  payload: state,
});
