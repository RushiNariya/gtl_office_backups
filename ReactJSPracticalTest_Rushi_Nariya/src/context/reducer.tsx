/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import {
  SEARCH_BY_CITY,
  SEARCH_CITY,
  SET_WEATHER_DATA,
  SET_ERROR,
  RESET_ERROR,
  REMOVE_CITY,
  REFRESH_STATE,
} from './actionTypes';
import { initState } from './InitialState';

const AppReducer = (state: initState, action: any): initState => {
  switch (action.type) {
    case SEARCH_BY_CITY:
      return {
        ...state,
      };
    case SET_WEATHER_DATA:
      return {
        ...state,
        data: action.payload,
      };
    case SEARCH_CITY:
      if (state.city.find((item) => item.name === action.payload.name)) {
        return state;
      }
      return {
        ...state,
        city: [...state.city, action.payload],
      };
    case REMOVE_CITY:
      return {
        ...state,
        data: {},
        city: state.city.filter((item) => item.name !== action.payload),
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    case RESET_ERROR:
      return {
        ...state,
        error: '',
      };
    case REFRESH_STATE:
      return {
        ...state,
        city: action.payload,
        error: '',
      };
    default:
      return state;
  }
};

export default AppReducer;
