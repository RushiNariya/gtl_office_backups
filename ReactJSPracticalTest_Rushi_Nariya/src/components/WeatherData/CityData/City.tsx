import React, { useState, useContext } from 'react';
import PropTypes from 'prop-types';
import './City.css';
import CityModal from './CityModal/CityModal';
import cityInfo from '../../../intefaces/cityInfo';
import { GlobalContext } from '../../../context/InitialState';

type props = {
  cityData: cityInfo;
};

function City({ cityData }: props): JSX.Element {
  const [show, setShow] = useState(false);

  function handleShow() {
    setShow(true);
  }

  const { removeCityHandler } =
    useContext(GlobalContext);
  const deleteSearchCity = (cityName: string) => {
    removeCityHandler(cityName);
  };
  return (
    <>
      <article className="city" onClick={() => handleShow()}>
        <div className="column-1">
          <div className="name">
            {cityData.name}
            <img
              className="flag"
              src={`http://openweathermap.org/img/wn/${cityData.icon}@2x.png`}
            />
          </div>
        </div>
        <div className="column-2">
          <div className="temp">
            <span className="units">{cityData.temp}</span>
            <span className="metrics">Â°C</span>
          </div>
        </div>
        <button
          className="cancel-button"
          onClick={() => deleteSearchCity(cityData.name)}
        >
          <p className="icon-cancel">x</p>
        </button>
      </article>
      {show && <CityModal show={show} setShow={setShow} cityData={cityData} />}
    </>
  );
}

City.propTypes = {
  cityData: PropTypes.shape({
    name: PropTypes.string.isRequired,
    temp: PropTypes.number.isRequired,
    icon: PropTypes.string.isRequired,
  }).isRequired,
};

export default City;
