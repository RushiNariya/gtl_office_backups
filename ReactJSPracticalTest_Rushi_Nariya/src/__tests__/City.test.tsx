/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import City from '../components/WeatherData/CityData/City';
import { GlobalProvider } from '../context/GlobalState';
import initialState from '../context/InitialState';
import cityInfo from '../intefaces/cityInfo';

let wrapped: ReactWrapper;

initialState.city = [
  {
    name: 'rajkot',
    temp: 32.04,
    icon: '50d',
    latitude: 21.0439,
    longitude: 74.5086,
    minTemp: 33.04,
    maxTemp: 31.04,
    country: 'IN',
  },
  {
    name: 'surat',
    temp: 32.04,
    icon: '50d',
    latitude: 21.0439,
    longitude: 74.5086,
    minTemp: 33.04,
    maxTemp: 31.04,
    country: 'IN',
  }
];

const city: cityInfo = {
  name: 'surat',
  temp: 32.04,
  icon: '50d',
  latitude: 21.0439,
  longitude: 74.5086,
  minTemp: 33.04,
  maxTemp: 31.04,
  country: 'IN',
};

beforeEach((): void => {
  wrapped = mount(
    <GlobalProvider initialState={initialState}>
      <City cityData={city} />
    </GlobalProvider>
  );
});

afterEach((): void => {
  wrapped.unmount();
});

describe('Test Case For City Component', (): void => {

  it('renders the City component', (): void => {
    expect(wrapped).toMatchSnapshot();
  });

  it('check article tag', (): void => {
    expect(wrapped.find('article').length).toEqual(1);
  });

  it('check correct data rendered in City Component ', (): void => {
    expect(wrapped.find('.units').text()).toEqual("32.04");
  });
});

describe('without click on city Component', (): void => {

  it('check cityModel does not render on without click', () => {
    expect(wrapped.find('CityModal').length).toEqual(0);
  });

});

describe('when click on city Component', (): void => {
  beforeEach(() => {
    wrapped
      .find('article')
      .simulate('click');
      wrapped.update();
  });

  it('check cityModel render on click', (): void => {
    expect(wrapped.find('CityModal').length).toEqual(1);
  });

});

describe('delete city from the list', (): void => {
  beforeEach(() => {
    wrapped
      .find('.cancel-button')
      .simulate('click');
      wrapped.update();
  });
  it('delete city on click', (): void => {
    expect(wrapped.find('.city').length).toEqual(1);
  })
});