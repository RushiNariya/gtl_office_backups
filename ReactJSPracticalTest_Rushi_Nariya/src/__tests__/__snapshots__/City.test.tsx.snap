// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case For AppTitle renders the City component 1`] = `
<GlobalProvider
  initialState={
    Object {
      "city": Array [
        Object {
          "country": "IN",
          "icon": "50d",
          "latitude": 21.0439,
          "longitude": 74.5086,
          "maxTemp": 31.04,
          "minTemp": 33.04,
          "name": "rajkot",
          "temp": 32.04,
        },
        Object {
          "country": "IN",
          "icon": "50d",
          "latitude": 21.0439,
          "longitude": 74.5086,
          "maxTemp": 31.04,
          "minTemp": 33.04,
          "name": "surat",
          "temp": 32.04,
        },
      ],
      "data": Object {},
      "error": "",
      "errorHandler": [Function],
      "removeCityHandler": [Function],
      "resetSetErrorHandler": [Function],
      "searchByCityHandler": [Function],
      "searchCityHandler": [Function],
      "setWeatherDataHandler": [Function],
    }
  }
>
  <City
    cityData={
      Object {
        "country": "IN",
        "icon": "50d",
        "latitude": 21.0439,
        "longitude": 74.5086,
        "maxTemp": 31.04,
        "minTemp": 33.04,
        "name": "surat",
        "temp": 32.04,
      }
    }
  >
    <article
      className="city"
      onClick={[Function]}
    >
      <div
        className="column-1"
      >
        <div
          className="name"
        >
          surat
          <img
            className="flag"
            src="http://openweathermap.org/img/wn/50d@2x.png"
          />
        </div>
      </div>
      <div
        className="column-2"
      >
        <div
          className="temp"
        >
          <span
            className="units"
          >
            32.04
          </span>
          <span
            className="metrics"
          >
            °C
          </span>
        </div>
      </div>
      <button
        className="cancel-button"
        onClick={[Function]}
      >
        <p
          className="icon-cancel"
        >
          x
        </p>
      </button>
    </article>
  </City>
</GlobalProvider>
`;

exports[`Test Case For City Component renders the City component 1`] = `
<GlobalProvider
  initialState={
    Object {
      "city": Array [
        Object {
          "country": "IN",
          "icon": "50d",
          "latitude": 21.0439,
          "longitude": 74.5086,
          "maxTemp": 31.04,
          "minTemp": 33.04,
          "name": "rajkot",
          "temp": 32.04,
        },
        Object {
          "country": "IN",
          "icon": "50d",
          "latitude": 21.0439,
          "longitude": 74.5086,
          "maxTemp": 31.04,
          "minTemp": 33.04,
          "name": "surat",
          "temp": 32.04,
        },
      ],
      "data": Object {},
      "error": "",
      "errorHandler": [Function],
      "removeCityHandler": [Function],
      "resetSetErrorHandler": [Function],
      "searchByCityHandler": [Function],
      "searchCityHandler": [Function],
      "setWeatherDataHandler": [Function],
    }
  }
>
  <City
    cityData={
      Object {
        "country": "IN",
        "icon": "50d",
        "latitude": 21.0439,
        "longitude": 74.5086,
        "maxTemp": 31.04,
        "minTemp": 33.04,
        "name": "surat",
        "temp": 32.04,
      }
    }
  >
    <article
      className="city"
      onClick={[Function]}
    >
      <div
        className="column-1"
      >
        <div
          className="name"
        >
          surat
          <img
            className="flag"
            src="http://openweathermap.org/img/wn/50d@2x.png"
          />
        </div>
      </div>
      <div
        className="column-2"
      >
        <div
          className="temp"
        >
          <span
            className="units"
          >
            32.04
          </span>
          <span
            className="metrics"
          >
            °C
          </span>
        </div>
      </div>
      <button
        className="cancel-button"
        onClick={[Function]}
      >
        <p
          className="icon-cancel"
        >
          x
        </p>
      </button>
    </article>
  </City>
</GlobalProvider>
`;
